src/blockrelay/blockrelay_common.cpp

```diff
@@ -177,36 +177,44 @@ bool ThinTypeRelay::AreTooManyBlocksInFlight()
 {
     // check if we've exceed the max thintype blocks in flight allowed.
     LOCK(cs_inflight);
-    if (mapThinTypeBlocksInFlight.size() >= MAX_THINTYPE_BLOCKS_IN_FLIGHT)
-        return true;
-    return false;
+    size_t mapSize = 0;
+    for (auto &entry : mapThinTypeBlocksInFlight)
+    {
+        // add the size of the sets of each entry
+        // it is possible for a set to be empty
+        mapSize = mapSize + entry.second.size();
+    }
+    return (mapSize >= MAX_THINTYPE_BLOCKS_IN_FLIGHT);
 }
 
 bool ThinTypeRelay::IsBlockInFlight(CNode *pfrom, const std::string thinType, const uint256 &hash)
 {
     // check if this node already has this thinType of block in flight.
     LOCK(cs_inflight);
-    auto range = mapThinTypeBlocksInFlight.equal_range(pfrom->GetId());
-    while (range.first != range.second)
+    auto key = mapThinTypeBlocksInFlight.find(pfrom->GetId());
+    if (key != mapThinTypeBlocksInFlight.end())
     {
-        if (range.first->second.thinType == thinType && range.first->second.hash == hash)
-            return true;
-
-        range.first++;
+        // time and recieved arent checked in the comparator, so they dont matter here
+        return key->second.count(CThinTypeBlockInFlight{hash, 0, false, thinType});
     }
     return false;
 }
 
 void ThinTypeRelay::BlockWasReceived(CNode *pfrom, const uint256 &hash)
 {
     LOCK(cs_inflight);
-    auto range = mapThinTypeBlocksInFlight.equal_range(pfrom->GetId());
-    while (range.first != range.second)
+    auto key = mapThinTypeBlocksInFlight.find(pfrom->GetId());
+    if (key != mapThinTypeBlocksInFlight.end())
     {
-        if (range.first->second.hash == hash)
-            range.first->second.fReceived = true;
-
-        range.first++;
+        for (auto entry : key->second)
+        {
```

Comment:
-auto entry
+auto &entry

Reply:
good catch. fixed

Reply:
good catch. thanks, fixed

Reply:
We cant do that here. adding & gives entry a const qualifier.